src/components/map/MapboxMap.tsx

```diff
@@ -0,0 +1,209 @@
+      if (error) throw error;
+      if (!map.hasImage(e.id)) map.addImage(e.id, image);
+    });
+  });
+
+  getEsriVectorSourceStyle(WHO_COUNTRY_VECTORTILES).then((style) => {
+    addEsriLayersFromVectorSourceStyle(style, map);
+    var styleJson = map.getStyle();
+
+
+    if (styleJson && styleJson.layers)
+    {
+      for (let layer of styleJson.layers) {
+        //@ts-ignore
+        const t = layer['source-layer'];
+        if (t === "DISPUTED_AREAS") {
+          map.moveLayer('Countries', layer.id); // HACK for now, moves countries layer behind border once loaded.
+          break;
+        }
+      }
+    }
+  });
+
+  let countryPop = new mapboxgl.Popup({ offset: 5, className: "pin-popup", closeButton: false });
+  let countryCode : string | undefined = undefined;
+
+  // Connect mouse movement events
+  map.on("mouseenter", "Countries", function (e : any) {
+    if (e.features[0].state.hasData)
+    {
+      countryPop
+      .setLngLat(e.lngLat)
+      .trackPointer()
+      .setHTML(ReactDOMServer.renderToString(CountryPopup(e.features[0], language)))
+      .addTo(map);
+    }
+  });
+
+  map.on("mouseleave", "Countries", function (e : any) {
+    if (countryPop.isOpen())
+    {
+      countryPop.remove()
+    }
+  });
+
+  map.on("mousemove", "Countries", function (e : any) {
+    if (e.features.length > 0 && e.features[0].id !== countryCode) {
+      if (e.features[0].state.hasData) {
+        countryPop.setHTML(ReactDOMServer.renderToString(CountryPopup(e.features[0], language)))
+        if(!countryPop.isOpen())
+        {
+          countryPop.addTo(map);
+        }
+      }
+      else
+      {
+        countryPop.remove()
+      }
+        countryCode = e.features[0].id
+      }
+    });
+
+  map.on("click", "study-pins", function (e: mapboxgl.MapMouseEvent & mapboxgl.EventData) {
+    const source_id = e.features[0].properties.source_id;
+
+    api.getRecordDetails(source_id).then((record) => {
+      if (record !== null){
+        new mapboxgl.Popup({ offset: 5, className: "pin-popup" })
+        .setLngLat(e.lngLat)
+        .setHTML(ReactDOMServer.renderToString(StudyPopup(record)))
+        .setMaxWidth("300px")
+        .addTo(map);
+      }
+    });
+  });
+
+  map.on("mouseenter", "study-pins", function () {
+    map.getCanvas().style.cursor = "pointer";
+  });
+  map.on("mouseleave", "study-pins", function () {
+    map.getCanvas().style.cursor = "";
+  });
+}
+
+
+
+const MapboxGLMap = () : any => {
+  const mapContainerRef = useRef<HTMLDivElement>(null);
+  const [state] = useContext(AppContext);
+  const api = new httpClient();
+  const mapContainer = useRef<mapboxgl.Map>(null);
+
+  // Creates map
+  useEffect(() => {
+    getEsriVectorSourceStyle(WHO_BASEMAP).then((baseMapStyle) => {
+      const map = new mapboxgl.Map({
+        //@ts-ignore
+        container: mapContainerRef.current,
+        style: baseMapStyle,
+        center: [-80, 45],
```

Comment:
Something to flag for the future, we may want to be able to supply the map center as a prop (e.g. if we want to have a serotracker.com/Explore/Canada)